addons/stock_delivery/models/stock_picking.py

```diff
@@ -71,12 +71,18 @@ def _compute_bulk_weight(self):
     @api.depends('move_line_ids.result_package_id', 'move_line_ids.result_package_id.shipping_weight', 'weight_bulk')
     def _compute_shipping_weight(self):
         for picking in self:
+            if picking.state == 'done':
+                continue
             # if shipping weight is not assigned => default to calculated product weight
             picking.shipping_weight = (
                 picking.weight_bulk +
                 sum(pack.shipping_weight or pack.weight for pack in picking.package_ids.sudo())
             )
 
+    def _action_done(self):
+        self._compute_shipping_weight()
```

Comment:
I don't like to call the compute manually, instead you could add 'state' in the depends

Reply:
if we do that 
state will already done when the compute is triggered 
but I want to skip that compute for done pickings 
that's why I opted for the solution of manually calling it before setting the picking as done

Reply:
we could exctrat this
``` py
            # if shipping weight is not assigned => default to calculated product weight
            picking.shipping_weight = (
                picking.weight_bulk +
                sum(pack.shipping_weight or pack.weight for pack in picking.package_ids.sudo())
            )
```
 into a seperate method and call it both in action_done and the compute but doesn't seem worth it for just one line

Reply:
I agree, it's not great to call the compute directly.
Also, this won't work as you'd expect if you don't set the pack's `shipping_weight` manually, which I expect to be the *vast majority* of the use cases. In this case, by the time you call the compute, quants won't be linked yet to the SML, meaning the package `weight` will be zero as well, forever setting the picking's `shipping_weight` to `0`.